docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,272 @@
+
+* JVM mode Vs Native mode
+  ** Quarkus is before all a Java framework, that means you can develop, package and run classic JAR application, that's what we call *JVM mode*.
+  ** Thanks to https://graalvm.org/[GraalVM] you can compile your Java application into machine specific code (like you do in Go or C++) and that's what we call *Native Mode*.
+
+The operation of compiling Java bytecode into a native system-specific machine code is named *Ahead of Time Compilation* (aka. AoT).
+
+* Build time Vs. Run time
+  ** The Build time corresponds to all the actions you apply to your Java source files to convert them into something runnable (class files, jar/war, native images). Usually this stage is composed by the compilation, annotation processing, bytecode generation, etc. At this point, everything is under developer's scope and control.
+  ** The Run time is all the actions that happens when you execute your application. It's obviously focused on starting your business-oriented actions but it relies on a lot of technical actions like loading libraries and configuration files, scanning the application's classpath, configuring the dependency injection, setting up your Object-Relationnal Mapping, instanciating your REST controllers, etc.
+
+Usually, Java frameworks do their bootstrapping during the Run time before actually starting your application "Business oriented layer". During the bootstrap frameworks dynamically collect metadata by scanning the classpath to find configurations, entity definitions, dependency injection binding, etc. in order to instanciate through reflection the proper obejcts. The main consequences are:
+
+* Delaying the readiness of your application: you need to wait couple of seconds before actually serve a business request.
+
+* Having a peak of resource consumption at bootstrap: In a constrained environment, you will need to size the needed resources based on your technical boostrap need rather than your actual business needs.
+
+Quarkus' pilosophy is to prevent as much as possible slow and memory intensive dynamic code execution by shifting left these actions and eventually do them during the build time . A Quarkus extension is a Java piece of code acting as an adapter layer for your favorite libary or technology.
+
+Description of a Quarkus extension
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A Quarkus extension consists of two parts:
+
+* Runtime module : Represent the capabilities the extension's developer exposes to the application's developer (an authentication filter, an enhanced data layer API, etc).
+
+* Deployment module: Used during the Augmentation Build time, it describes how to "deploy" a framework
+following Quarkus philosophy. In other words, it applies all the Quarkus optimization to yoir application before actually running it.
+
+At this point, you should have understand that most of the magic will happen at the Augmentation Build time thanks to the deployment module.
```

Comment:
Just a note. The _Augmentation Build time_ is referred to as a _deployment time_ in other documents. Using different name for the same thing may be confusing to the readers so maybe we should add some explanation here, but I agree that "augmentation time" is a better term since "deployment time" can be easily confused with a process of deploying service to the server.

Reply:
I would just say "at build time". And btw "you should have understood".

Reply:
IMHO, deployment/augmentation is a step of the build time (just as compilation is one). My 2cts, only use "build time" hides too much subtle details

Good catch s/understand/understood

Reply:
I don't have a strong opinion here. It's now down to detailing build phases which may be great to know but also could look like too much info for users trying to get their first extension working. If you want to keep "Augmentation" I'd prefer "augmentation phase of the build" though. Thanks again for taking care of this guide.

Reply:
I replaced this part as asked